fenix/app/src/main/java/org/mozilla/fenix/compose/PagerIndicator.kt

```diff
@@ -0,0 +1,132 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.pager.HorizontalPager
+import com.google.accompanist.pager.PagerState
+import com.google.accompanist.pager.rememberPagerState
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.theme.FirefoxTheme
+
+private val DefaultIndicatorSize = 6.dp
+private val DefaultIndicatorSpacing = 8.dp
+private val DefaultIndicatorShape = CircleShape
+
+/**
+ * An horizontally laid out indicator for a [HorizontalPager] with the ability to leave the trail of
+ * indicators to show progress, instead of just showing the current one as active.
+ *
+ * @param pagerState The state object of your [HorizontalPager] to be used to observe the list's state.
+ * @param modifier The modifier to apply to this layout.
+ * @param pageCount The size of indicators should be displayed, defaults to [PagerState.pageCount].
+ * If you are implementing a looping pager with a much larger [PagerState.pageCount]
+ * than indicators should displayed, e.g. [Int.MAX_VALUE], specify you real size in this param.
+ * @param activeColor The color of the active page indicator, and the color of previous page
+ * indicators in case [leaveTrail] is set to true.
+ * @param inactiveColor The color of page indicators that are inactive.
+ * @param leaveTrail Whether to leave the trail of indicators to show progress.
+ * This defaults to false and just shows the current one as active.
+ */
+
+@Composable
+fun PagerIndicator(
+    pagerState: PagerState,
+    modifier: Modifier = Modifier,
+    pageCount: Int = pagerState.pageCount,
+    activeColor: Color,
+    inactiveColor: Color,
+    leaveTrail: Boolean = false,
+) {
+    Row(
+        modifier = modifier,
+        horizontalArrangement = Arrangement.spacedBy(DefaultIndicatorSpacing),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        val basicIndicatorModifier = Modifier.size(DefaultIndicatorSize)
+        val inactiveModifier = basicIndicatorModifier
+            .background(color = inactiveColor, shape = DefaultIndicatorShape)
+        val activeModifier = basicIndicatorModifier
+            .background(color = activeColor, shape = DefaultIndicatorShape)
+
+        val conditionForActiveModifier: (pageIndex: Int) -> Boolean =
+            if (leaveTrail) {
+                { it <= pagerState.currentPage }
+            } else {
+                { it == pagerState.currentPage }
+            }
+
+        repeat(pageCount) {
+            if (conditionForActiveModifier(it)) {
+                Box(activeModifier)
+            } else {
+                Box(inactiveModifier)
+            }
+        }
```

Comment:
```suggestion
        val showActiveModifier: (pageIndex: Int) -> Boolean =
            if (leaveTrail) {
                { it <= pagerState.currentPage }
            } else {
                { it == pagerState.currentPage }
            }

        repeat(pageCount) {
            Box(modifier = Modifier
                .size(6.dp)
                .background(
                    color = if (showActiveModifier(it) {
                        activeColor                  
                    } else {
                        inactiveColor
                   },
                    shape = CircleShape
                )
            )
        }
```
I think we could simplify this much further since the only thing that changes between these modifier variables is the color so we could avoid the extra variables and if statement checks by inlining. I also think we can just inline the size, spacing and shape here and remove the need of the constants. Part of my thinking here is to really move away from the prior accompanist code.

I think you improve upon my sample suggestion by changing the function to return the indicator color instead of doing the if statement check here.

Reply:
Fair point, updated and that removed the need for constants 👍  Left the if condition as creating a function required passing in the colors as params anyway.